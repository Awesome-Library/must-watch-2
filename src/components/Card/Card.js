import React from 'react';
import PropTypes from 'prop-types';

let Card = (props) => {
  let {title, overview, poster} = props;

  return (
    <div className="card">
      <div className="card-image">
        <figure className="image is-4by2">
          <img src={poster} alt="" />
        </figure>
      </div>
      <div className="card-content" style={{height: 200}}>
        <div className="media">
          <div className="media-content">
            <p className="title is-4">
              {title}
            </p>
          </div>
        </div>

        <div className="content">
          <div className="columns">
            <div className="column">
              <p style={{minHeight: 90}}>
                {overview.slice(0, 70)} ...
              </p>
            </div>
          </div>
        </div>
      </div>

      <footer className="card-footer">
        <a className="card-footer-item">
          Adicionar a lista
        </a>
      </footer>
    </div>
  );
}

Card.propTypes = {
  title: PropTypes.string.isRequired,
  overview: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired
};

export default Card;
