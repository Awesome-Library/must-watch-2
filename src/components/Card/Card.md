Card example:

```example
<div style={{width: 300}}>
  <Card
    title="pedilari"
    overview="lorem ipsum dolor sit amet consecteur lorem"
    poster="http://placehold.it/350x500" />
</div>
```
