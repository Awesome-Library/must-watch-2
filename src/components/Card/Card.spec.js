import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';

it('renders Card', () => {
  const tree = renderer.create(
    <Card
      title="pedilari"
      overview="lorem ipsum dolor sit amet consecteur lorem"
      poster="http://placehold.it/350x350" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
