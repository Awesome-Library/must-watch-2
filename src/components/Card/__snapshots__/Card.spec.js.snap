exports[`test renders Card 1`] = `
<div
  className="card">
  <div
    className="card-image">
    <figure
      className="image is-4by2">
      <img
        alt=""
        src="http://placehold.it/350x350" />
    </figure>
  </div>
  <div
    className="card-content"
    style={
      Object {
        "height": 200,
      }
    }>
    <div
      className="media">
      <div
        className="media-content">
        <p
          className="title is-4">
          pedilari
        </p>
      </div>
    </div>
    <div
      className="content">
      <div
        className="columns">
        <div
          className="column">
          <p
            style={
              Object {
                "minHeight": 90,
              }
            }>
            lorem ipsum dolor sit amet consecteur lorem
             ...
          </p>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="card-footer">
    <a
      className="card-footer-item">
      Adicionar a lista
    </a>
  </footer>
</div>
`;
